spring-boot-project/spring-boot-docs/src/main/asciidoc/spring-boot-features.adoc

```diff
@@ -4228,72 +4228,90 @@ the Mongo instance's configuration and logging routing.
-bean:
+To access a Neo4j server, you can inject an auto-configured `org.neo4j.ogm.session.Session`. 
+By default, the instance tries to connect to a Neo4j server at `localhost:7687` via the Bolt 
+protocol. The following example shows how to inject a Neo4j session:
 
 [source,java,indent=0]
 ----
 	@Component
 	public class MyBean {
 
-		private final Neo4jTemplate neo4jTemplate;
+		private final Session session;
 
 		@Autowired
-		public MyBean(Neo4jTemplate neo4jTemplate) {
-			this.neo4jTemplate = neo4jTemplate;
+		public MyBean(Session session) {
+			this.session = session;
 		}
 
 		// ...
 
 	}
 ----
 
-You can take full control of the configuration by adding a
-`org.neo4j.ogm.config.Configuration` `@Bean` of your own. Also, adding a `@Bean` of type
-`Neo4jOperations` disables the auto-configuration.
-
-You can configure the user and credentials to use by setting the `spring.data.neo4j.*`
+You can configure the uri and credentials to use by setting the `spring.data.neo4j.*`
 properties, as shown in the following example:
 
 [source,properties,indent=0]
 ----
-	spring.data.neo4j.uri=http://my-server:7474
+	spring.data.neo4j.uri=bolt://my-server:7687
 	spring.data.neo4j.username=neo4j
 	spring.data.neo4j.password=secret
 ----
 
+You can take full control over the session creation by adding a 
+`org.neo4j.ogm.config.Configuration` `@Bean`. To disable the auto-configuration
+provided by the "`Starter`" completely, add a `@Bean` of type 
+`org.neo4j.ogm.session.SessionFactory`. Take note that you have to provide a
+`org.springframework.data.neo4j.transaction.Neo4jTransactionManager` as well.
 
 
 [[boot-features-connecting-to-neo4j-embedded]]
 ==== Using the Embedded Mode
 
 If you add `org.neo4j:neo4j-ogm-embedded-driver` to the dependencies of your application,
 Spring Boot automatically configures an in-process embedded instance of Neo4j that does
-not persist any data when your application shuts down. You can explicitly disable that
-mode by setting `spring.data.neo4j.embedded.enabled=false`. You can also enable
-persistence for the embedded mode by providing a path to a database file, as shown in the
-following example:
+not persist any data when your application shuts down. As the embedded Neo4j OGM  driver 
+does not provide the Neo4j kernel itself, you have to provide this dependency manually. 
+Refer to https://neo4j.com/docs/ogm-manual/current/reference/#reference:getting-started[the
+Neo4j OGM documentation] for list of compatible versions.
+
+Your dependencies for using embedded Neo4j should look as follows:
 
+[source,xml,indent=0]
 ----
-	spring.data.neo4j.uri=file://var/tmp/graph.db
+	<dependency>
+		<groupId>org.neo4j</groupId>
+		<artifactId>neo4j-ogm-embedded-driver</artifactId>
+		<version>${neo4j-ogm.version}</version>
+	</dependency>
+	<dependency>
+		<groupId>org.neo4j</groupId>
+		<artifactId>neo4j</artifactId>
+		<version>3.4.7</version>
```

Comment:
Sorry but I am not keen to hard-code versions in our doc. I think the link that is referenced above should mention this (if it doesn't already).